packages/vitest/src/node/create.ts

```diff
@@ -11,10 +12,16 @@ export async function createVitest(mode: VitestRunMode, options: UserConfig, vit
   const ctx = new Vitest(mode)
   const root = resolve(options.root || process.cwd())
 
-  const configPath = options.config
+  let configPath = options.config
     ? resolve(root, options.config)
     : await findUp(configFiles, { cwd: root } as any)
 
+  let jsonConfig = {}
+  if (configPath && configPath.endsWith('json')) {
+    jsonConfig = JSON.parse(readFileSync(configPath, 'utf-8'))
+    configPath = undefined
```

Comment:
This will use several configs (json and vite.config), if you have separate configs for Vite and Vitest, because undefined `configPath` will force Vite to use close one.

It's better to add this support to Vite directly, which doesn't really make sense, because you can't add plugins there.